### PR TITLE
[FLINK-18968] Translate README.md to Chinese 

### PR DESCRIPTION
**Pull request by billyrrr**
Translate README.md to Chinese. New file is named README.zh.md 
Methodology: Google Translate on HTML + Manual Correction
To quickly check diff: click 22ed8a9

### CODE REVIEW

File: README.zh.md
Comment (billyrrr) [inline README.zh.md:140]: TODO: fix 构建项目

**Comment by igalshilman** [comment]
Thanks for your contribution @billyrrr!
Unfortunately I don't speak Chinese so it would be a little difficult to review this.
I would try to find someone from the Flink community that can help with the review.

File: README.zh.md
Comment (carp84) [inline README.zh.md:30]:
```suggestion
- [参与贡献](#contributing) 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:38]:
```suggestion
Stateful Functions 应用程序由以下原语组成：有状态函数，入口(ingress)，路由器(router)和出口(egress)。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:63]:
```suggestion
在于，函数不是在定义数据流（拓扑）的有向无环图（DAG）中组装，
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:70]:
```suggestion
入口可以是消息队列，日志或HTTP服务器-它们产生事件并交由应用程序处理。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:72]:
```suggestion
* _路由器_(Router)连接到入口(Ingress)并决定最先处理事件的函数实例。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:76]:
```suggestion
出口是可选的；也有可能没有事件离开应用程序，函数会完成事件的处理或直接调用其他应用程序。 
```

**Review by carp84 — CHANGES_REQUESTED** [review]
Thanks for the contribution @billyrrr ! Please check my first-round review comments, I will continue and complete the review in the next round.

**Comment by billyrrr** [comment]
Hi, @carp84 Thank you for the review. 

I think that "stateful functions" should be translated to chinese to promote communications within people who primarily speak Chinese. Suppose that someone is proficient at processing stateful streams but not proficient in English, they would not be able to notice the target use case of Flink Stateful Functions. On another note, when someone wants to refer flink stateful functions to another person, it is easier to use the term 'Flink有状态函数' in their native language rather than ' Flink Stateful Functions'. Nevertheless, the English term on its own has its merit of accuracy and ease of finding related documentations. So I do understand your point too. 

I think that there are more benefit from translating 'Flink Stateful Functions' to 'Flink有状态函数'. Can we discuss more about this decision? Thanks!

**Comment by carp84** [comment]
Thanks for sharing the thoughts @billyrrr , and sure let's discuss more about it (and sorry for the late response, busy preparing for this year's Singles' Day recently).

I think translating the name into Chinese will cause confusions in some cases, such as the below one:
`A Stateful Functions application consists of the following primitives: stateful functions, ingresses, routers and egresses`
-> `有状态函数应用程序由以下原语组成：有状态函数，入口(ingress)，路由器(router)和出口(egress)` 
Comparing to:
-> `Stateful Functions 应用程序由以下原语组成：有状态函数，入口(ingress)，路由器(router)和出口(egress)`
What do you think?

And in Flink we will also keep the original names for some terms such as `State Backend` (referring to [Flink's translation specification](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications))

PS. Maybe we should have a similar translation specification for stateful functions? Wdyt? @igalshilman @tzulitai Thanks.

File: README.zh.md
Comment (carp84) [inline README.zh.md:103]:
```suggestion
* __语言无关性：__函数调用使用一个简单的基于HTTP/gRPC的协议，因此可以用各种语言轻松地实现函数。
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:105]:
```suggestion
这使得在Kubernetes和FaaS平台上，或（微）服务后台执行函数，同时在函数之间提供一致的状态和轻量级消息传递成为可能。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:109]:
```suggestion
按照下面的步骤即可立刻开始使用 Stateful Functions。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:112]:
```suggestion
本指南将引导您通过设置开始开发和测试自己的 Stateful Functions（Java）应用程序，并运行一个示例。如果您想使用Python快速开始，
请查看[StateFun Python SDK](https://github.com/apache/flink-statefun/tree/master/statefun-python-sdk)和[Python Greeter示例](https://github.com/apache/flink-statefun/tree/master/statefun-examples/statefun-python-greeter-example) 。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:124]:
```suggestion
您可以使用下面的Maven命令快速开始构建 Stateful Functions 应用程序： 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:134]:
```suggestion
这使您可以命名新创建的项目。它将以交互方式询问您`` GroupId `` ， 
`` ArtifactId ``和 package 名称，并将生成一个与您指定的`` ArtifactId ``同名的新目录。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:138]:
```suggestion
IntelliJ IDEA天然支持Maven项目。如果使用Eclipse，则需要使用`` m2e ``插件导入
Maven项目。一些 Eclipse发布版本默认包含该插件，而另一些则需要您手动安装。 
```

File: README.zh.md
Comment (carp84) [reply README.zh.md:140]: I think `构建项目` is better.

File: README.zh.md
Comment (carp84) [inline README.zh.md:142]:
```suggestion
如果要构建/打包项目，请进入项目目录并运行`` mvn clean package ``命令。您将找到一个包含您的应用程序以及相关依赖的JAR包：`target/<artifact-id>-<version>.jar`。 
```

**Comment by tzulitai** [comment]
A similar translation specification for StateFun is indeed nice, especially if we want to move along with this, it would be best to have consensus on what not to translate.

IMHO, I agree with @carp84 here, not translating the name of the project makes a lot of sense here.

File: README.zh.md
Comment (carp84) [inline README.zh.md:146]:
```suggestion
要测试您的应用程序，可以直接在IDE中运行它，而无需进一步的打包或部署。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:152]:
```suggestion
作为一个简单的演示，我们将逐步运行[Greeter示例](statefun-examples/statefun-greeter-example)。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:165]:
```suggestion
要查看实际的运行情况，请向topic `` names ``发送一些消息，并查看topic `` greetings `` 的输出： 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:183]:
```suggestion
Stateful Functions 应用程序可以打包为[独立应用程序](https://ci.apache.org/projects/flink/flink-statefun-docs-master/deployment-and-operations/packaging.html#images)或[Flink作业](https://ci.apache.org/projects/flink/flink-statefun-docs-master/deployment-and-operations/packaging.html#flink-jar) ，并提交给Flink集群运行。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:187]:
```suggestion
以下是一个Dockerfile示例，用于为名为`` statefun-example ``的应用程序构建带有[嵌入式](https://ci.apache.org/projects/flink/flink-statefun-docs-master/sdk/modules.html#embedded-module)模块（Java）的 Stateful Functions 镜像。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:199]:
```suggestion
如果您希望将 Stateful Functions 应用程序打包为 Flink 作业以提交到现有的Flink集群，只需在你的应用程序中将`` statefun-flink-distribution ``添加为依赖项。 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:213]:
```suggestion
  <strong>注意:</strong>该发行版必须捆绑在你的应用程序fat JAR中，以将它放置于Flink的[用户代码类加载器上](https://ci.apache.org/projects/flink/flink-docs-stable/monitoring/debugging_classloading.html#inverted-class-loading-and-classloader-resolution-order) 
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:220]:
```suggestion
## <a name="contributing"></a>参与贡献
```

File: README.zh.md
Comment (carp84) [inline README.zh.md:224]:
```suggestion
您可以在[Apache Flink网站](https://flink.apache.org/contributing/how-to-contribute.html)上了解关于如何做出贡献的更多信息。对于代码贡献，请仔细阅读“ [贡献代码”](https://flink.apache.org/contributing/contribute-code.html)部分，并检查[Jira](https://issues.apache.org/jira/browse/FLINK-15969?jql=project%20%3D%20FLINK%20AND%20component%20%3D%20%22Stateful%20Functions%22)中的_Stateful Functions_组件以概要了解正在进行中的社区工作。 
```

**Review by carp84 — CHANGES_REQUESTED** [review]
2nd round review. We could do adjustment accordingly after getting conclusion on whether to translate the name of `Stateful Functions`.

**Comment by tzulitai** [comment]
Just to make it a bit more clearer:
I think before we merge this, it is best if we can extend the [Flink translation specifications](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) for Stateful Functions first.
That would set a better foundation for future contributions on further translating the documentations.

@billyrrr do you think that would be possible? Or do you prefer merging this first, and work on the specifications afterwards?

**Comment by billyrrr** [comment]
@tzulitai I can update the specifications first

File: README.zh.md
Comment (billyrrr) [reply README.zh.md:8]: I agree with the translations, but I think it is best preserve line breaks; so that when the corresponding English docs changes, the relevant line in Chinese docs can be quickly located. This also assist with machine learning translations, as a training set for translations can be easily extracted.

File: README.zh.md
Comment (billyrrr) [reply README.zh.md:19]: 这里应该是在说 Stateful Functions, ingress, router, egress 是如何将更底层的Flink操作抽象化。 所以翻译成```抽象```可能更好一点

File: README.zh.md
Comment (billyrrr) [reply README.zh.md:63]:
```suggestion
在于，函数不是在定义数据流动（流的拓扑结构）的有向非循环图（DAG）中组装，
```

**Comment by carp84** [comment]
@klion26 it would be great if you could also give some suggestions about translation specification here with your experience, thanks!

**Comment by klion26** [comment]
@carp84 thanks for the notice. will review this late today.

File: README.zh.md
Comment (klion26) [reply README.zh.md:134]: +1

File: README.zh.md
Comment (klion26) [reply README.zh.md:146]: `可以直接在 IDE 中运行测试你的程序，无需进一步打包或部署`

File: README.zh.md
Comment (klion26) [inline README.zh.md:148]: `详情可以参考 [Harness 示例]`

File: README.zh.md
Comment (klion26) [inline README.zh.md:222]: Personally, I don't think `运行时和操作` is a good translation for  `the runtime and operations`, because people can't get the meaning easily.

**Review by klion26 — COMMENTED** [review]
I had a round of review for the translation, and left some comments.

Aside from the change, I have a question: how can the user reach the rendered page of the Chinese version for the current doc?

**Comment by carp84** [comment]
> Just to make it a bit more clearer:
> I think before we merge this, it is best if we can extend the [Flink translation specifications](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) for Stateful Functions first.
> That would set a better foundation for future contributions on further translating the documentations.

Thanks for helping review @klion26 , but I meant to ask whether you could help forming the translation specification for Stateful Functions according to your experience in Flink document translation (smile).

**Comment by klion26** [comment]
@carp84 got it. It's my pleasure to help to form the glossary for Stateful Functions(I think we can reuse the other parts). I'll first draft a glossary for Stateful Functions according to my understanding.

**Comment by klion26** [comment]
Sorry for the late reply, dealt with some urgent things on hand. 
I've drafted a [Flink Stateful Functions Translation Specifications](https://docs.google.com/document/d/1Fv56PG50pHlDxZfF49iRhv-8dY7ZaCSHUCvBM2Vy0ao/edit?usp=sharing), please take a look when you're free, thanks.
cc @tzulitai @carp84 @billyrrr

**Comment by carp84** [comment]
> Sorry for the late reply, dealt with some urgent things on hand.
> I've drafted a [Flink Stateful Functions Translation Specifications](https://docs.google.com/document/d/1Fv56PG50pHlDxZfF49iRhv-8dY7ZaCSHUCvBM2Vy0ao/edit?usp=sharing), please take a look when you're free, thanks.
> cc @tzulitai @carp84 @billyrrr

Thanks for driving this and sorry for the late reply @klion26 . The draft LGTM and I've left some minor comments on google doc.

@tzulitai IMHO we're ready to start a discussion in ML, wdyt? Thanks.

**Comment by klion26** [comment]
@carp84 thanks for the reply and comments. will check it now.

**Comment by tzulitai** [comment]
@carp84 @klion26 
Thanks for drafting up the translation specifications and driving this!
I took a look, and overall it looks good to me. Yes, I think we are ready to propose a discussion to the ML lists now.

**Comment by klion26** [comment]
FYI, The Translation Specification for Stateful Functions has been added to the [existing wiki](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications)

**Comment by carp84** [comment]
Thanks for the great job @klion26 !

@billyrrr could you update the PR according to the review suggestions and [translation specification](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications)? Thanks.

**Comment by tzulitai** [comment]
@billyrrr do you plan on updating this PR?

Or alternatively, @klion26 do you have in mind anyone who would be suitable for taking this translation over if @billyrrr remains absent?

**Comment by klion26** [comment]
@wangzzu Do you want to take over this?

**Comment by wangzzu** [comment]
> @wangzzu Do you want to take over this?

thanks for inviting, I'd be pleased to continue @billyrrr's  previous work, I will modify it according to comments  as soon as possible.

**Comment by tzulitai** [comment]
Thanks @wangzzu!
Please remember to retain @billyrrr original commits, and consider opening a separate PR with your commits on top.

**Comment by klion26** [comment]
@wangzzu is there any update?

**Comment by wangzzu** [comment]
@tzulitai @klion26 Sorry for the late reply, I will open a new PR soon.

**Comment by wangzzu** [comment]
@tzulitai @klion26 I have submitted the new PR: https://github.com/apache/flink-statefun/pull/176

**Comment by carp84** [comment]
Closing this PR since it's already superseded by #176

**Pull request closed without merging**
